### PR TITLE
Adjust the hyperopt and sigopt as extras requirements

### PR DESCRIPTION
**Pull request by yiliu30**
Signed-off-by: yiliu30 <yi4.liu@intel.com>

## Type of Change

- bug fix
- API changed or not: None

## Description

Adjust the `hyperopt` and `sigopt` as extras requirements.
JIRA ticket: [ILITV-2452](https://jira.devtools.intel.com/browse/ILITV-2452)

## Expected Behavior & Potential Risk
`hyperopt` and `sigopt`  are not hard dependencies.

## How has this PR been tested?

Pre-CI

## Dependency Change?

Any library dependency introduced or removed:
- `hyperopt` is required only when the user uses `tpe` strategy.
- `sigopt` is required only when the user uses `sigopt` strategy.

### CODE REVIEW

**Comment by yiliu30** [comment]
The Pre-CI test failed with a performance regression which may be a random issue caused by the differences in test nodes.

image

**Comment by chensuyue** [comment]
> The Pre-CI test failed with a performance regression which may be a random issue caused by the differences in test nodes.
> 
> image

Yes, you are right.

**Review by chensuyue — APPROVED** [review]

**Review by intel-zhangyi — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -36,8 +36,8 @@
 
 # define install requirements
 install_requires_list = [
-        'numpy', 'pyyaml', 'scikit-learn', 'schema', 'py-cpuinfo', 'hyperopt', 'pandas', 'pycocotools',
-        'opencv-python', 'requests', 'psutil', 'Pillow', 'sigopt', 'prettytable', 'cryptography', 'Cython',
+        'numpy', 'pyyaml', 'scikit-learn', 'schema', 'py-cpuinfo', 'pandas', 'pycocotools',
+        'opencv-python', 'requests', 'psutil', 'Pillow', 'prettytable', 'cryptography', 'Cython',
```
Comment (ftian1) [inline setup.py:40]: cython should be removed. pls confirm with suyue

File: requirements.txt [hunk 2]
```diff
@@ -20,7 +19,6 @@ Pillow
 pycocotools-windows; sys_platform != 'linux'
 pycocotools; sys_platform == 'linux'
 opencv-python
-sigopt
```
Comment (ftian1) [inline requirements.txt:23]: check cython

File: setup.py (hunk 1 above)
Comment (yiliu30) [reply setup.py:40]: Hi Feng, have confirmed with @chensuyue that she will remove `Cpython` by another PR.

File: requirements.txt (hunk 2 above)
Comment (chensuyue) [reply requirements.txt:23]: I will handle the other 2 dependencies.

**Pull request merged**